server/server.py

```diff
@@ -384,6 +385,11 @@ def do_GET(self):
             except:
                 message = json.dumps({"Error": "Syntax error in set command URL.", "url": self.path})
                 log.debug("Syntax error in set command URL: %s" % self.path)
+            if(id not in deviceslist):
+                for key in deviceslist:
+                    if deviceslist[key]['name'] == urllib.parse.unquote(id):
+                        id = deviceslist[key]['id']
+                        break
```

Comment:
Minor note: Since we are repeating this same logic 6 times, should we consider making this a function?

Reply:
Yes, of course. I was unsure to put it into a function... I will change it accordingly.

Reply:
I would also pull the `urllib.parse.unquote(id)` out of the loop and move it to a variable beforehand instead.

Reply:
I included both suggestions: getDeviceIdByName as a function and pull `urllib.parse.unquote(id)` out of the loop.